snapcraft/tests/fixture_setup.py

```diff
@@ -875,3 +877,57 @@ def fetch_binary(self, dir_, progress):
 
     def get_dependencies(self, _):
         return []
+
+
+class WithSnapNotInstalled(fixtures.Fixture):
```

Comment:
My first guess from the name was that it's mocking that the snap is not installed, cf. `FakeSnapd`. Seeing that it's actually asserting that the snap isn't installed yet, it would obviously have to be an `assertSnapNotInstalled` rather than a fixture. Then I realized it's also removing the snap in the cleanup, which explains why you went for a fixture... to my mind, `SnapInstalledDuringTheTest` would convey better what it's actually doing.

Reply:
I don't like either, but prefer @elopio's... @kyrofa you are the tie breaker!

Reply:
Yeah there's a lot going on here that's difficult to convey in a simple class name. Where's my docstring?!

Let me think this through:

- This fixture doesn't care that the snap was actually installed during the test. Indeed, `snap remove nonexisting-snap` exits 0. As such, `SnapInstalledDuringTheTest` does not make sense.
- This fixture DOES care if the snap is installed before the test by failing in that case. So @elopio's direction is the right one.

So the real question is: which is more important? The assertion at the beginning, or the cleanup step? I think it's more important that it fails is the snap was already installed, because otherwise the test isn't testing the right thing. Perhaps `WithoutSnapInstalled` would be better, though?